Sources/Netswift/NetswiftPerformer.swift

```diff
@@ -39,4 +39,28 @@ open class NetswiftPerformer: NetswiftNetworkPerformer {
         }
         return nil
     }
+
+    @available(iOS 15.0.0, *)
+    open func perform<T: NetswiftRequest>(_ request: T) async -> NetswiftResult<T.Response> {
+        await withCheckedContinuation { continuation in
+            switch request.serialise() {
+            case .success(let url):
+                _ = self.requestPerformer.perform(url) { response in
+
+                    switch response {
+                    case .success:
+                        let networkResponse = response
+                            .flatMap(request.decode)
+                            .flatMap(request.cast)
+                            .flatMap(request.deserialise)
+                        continuation.resume(returning: networkResponse)
+                    case .failure(let error):
+                        continuation.resume(returning: .failure(error))
+                    }
+                }
+            case .failure(let error):
+                continuation.resume(returning: .failure(error))
+            }
+        }
```

Comment:
I'm confused here as to why you didn't actually use the `async` HTTPPerformer api you just implemented? 🤔 
```suggestion
        switch request.serialise() {
        case .success(let url):
            switch await requestPerformer.perform(url) {
            case .success:
                return response
                    .flatMap(request.decode)
                    .flatMap(request.cast)
                    .flatMap(request.deserialise)
                    
                case .failure(let error):
                    return .failure(error)
            }
            
        case .failure(let error):
            return .failure(error)
        }
```